src/coreclr/jit/lsra.cpp

```diff
@@ -9530,11 +9547,15 @@ void LinearScan::TupleStyleDump(LsraTupleDumpMode mode)
         {
             bool printedBlockHeader = false;
             // We should find the boundary RefPositions in the order of exposed uses, dummy defs, and the blocks
-            for (; refPosIterator != refPositions.end() &&
-                   (currentRefPosition->refType == RefTypeExpUse || currentRefPosition->refType == RefTypeDummyDef ||
-                    (currentRefPosition->refType == RefTypeBB && !printedBlockHeader));
-                 ++refPosIterator, currentRefPosition = &refPosIterator)
+            for (; refPosIterator != refPositions.end(); ++refPosIterator)
             {
+                currentRefPosition = &refPosIterator;
+                if (currentRefPosition->refType != RefTypeExpUse && currentRefPosition->refType != RefTypeDummyDef &&
+                    !(currentRefPosition->refType == RefTypeBB && !printedBlockHeader))
```

Comment:
This particular inversion of a clause leaves a double negative, though arguably it's the "right" way to express "not a RefTypeBB as long as we haven't printed one yet".  In other words, `(currentRefPosition->refType != RefTypeBB || printedBlockHeader)` seems more confusing.

At the risk of dragging you into more cleanup (or maybe I could just fold it into my follow-up?), maybe the loop condition could just be dropped.  Then below the `RefTypeBB` case could do `if (printedBlockHeader) break;` and `default` could drop the `printf` and just `break`.  Unfortunately, `break` in both of those means breaking out of the for loop, not the switch.

I'll post this and go finish reading the diffs.  I'm leaning towards doing anything as follow-up as the complexity of the original clause has nothing to do with this PR's purpose.

Reply:
Yeah, I'm fine either way on this one for this PR.  Other than the one I commented one, the others are definitely improvements.  Thanks!